pygmt/src/solar.py

```diff
@@ -22,7 +26,12 @@
     t="transparency",
 )
 @kwargs_to_strings(R="sequence", c="sequence_comma", p="sequence")
-def solar(self, terminator="d", terminator_datetime=None, **kwargs):
+def solar(
+    self,
+    terminator: Literal["day_night", "civil", "nautical", "astronomical"] = "day_night",
+    terminator_datetime=None,
+    **kwargs,
+):
```

Comment:
Maybe we want to sort the types of terminators alphabetically?
```suggestion
def solar(
    self,
    terminator: Literal["astronomical","civil","day_night","nautical"] = "day_night",
    terminator_datetime=None,
    **kwargs,
):
```